app/src/main/java/com/ics342/weatherapp/MyAdapter.kt

```diff
@@ -16,72 +18,107 @@ class MyAdapter(private val data: List<DayForecast>) :
         }
 
         fun bindSunset(data: DayForecast) {
-            // THIS IS FOR NEXT ASSIGNMENT
-            //val instant = Instant.ofEpochSecond(data.sunset)
-            //val dateTime = LocalDateTime.ofInstant(instant, ZoneId.systemDefault())
-            //sunsetView.text = formatter2.format(dateTime)
+            val instant = Instant.ofEpochSecond(data.sunset)
+            val dateTime = LocalDateTime.ofInstant(instant, ZoneId.systemDefault())
+            val hours = formatter2.format(dateTime).toInt()   // h
+            val min = formatter3.format(dateTime).toInt()  // mm
 
-            sunsetView.text = data.sunset   // Used this for hard coded
+            sunsetView.text = sunsetView.context.getString(R.string.sunset, hours, min)
         }
 
-        fun bindTemp(data: DayForecast) {
-            // THIS IS FOR NEXT ASSIGNMENT
-            //val instant = Instant.ofEpochSecond(data.temp)
-            tempView.text = data.temp    // Used this for hard coded
+        // Bind for day temperature, Temp
+        fun bindTempDay(data: DayForecast) {
+            tempView.text =
+                tempView.context.getString(R.string.temp, data.temp.day.toInt())
         }
 
-        fun bindHigh(data: DayForecast) {
-            // THIS IS FOR NEXT ASSIGNMENT
-            //val instant = Instant.ofEpochSecond(data.temp)
-            highView.text = data.high    // Used this for hard coded
+        // Bind for max temperature, Max/High
+        fun bindTempMax(data: DayForecast) {
+            maxView.text = maxView.context.getString(R.string.max, data.temp.max.toInt())
         }
 
-        fun bindLow(data: DayForecast) {
-            // THIS IS FOR NEXT ASSIGNMENT
-            //val instant = Instant.ofEpochSecond(data.temp)
-            lowView.text = data.low    // Used this for hard coded
+        // Bind for min temperature, Min/Low
+        fun bindTempMin(data: DayForecast) {
+            minView.text = minView.context.getString(R.string.min, data.temp.min.toInt())
         }
     }
 
+    /**
+     * This is an Important Method for RecyclerView, called onCreateViewHolder.
+     *
+     * This inflates instance of row_date.xml, This means it reuses the view
+     * over and over.
+     *
+     * Called by RecyclerView when it needs a new view holder of the specified type,
+     * second parameter is type
+     */
     override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
         val view = LayoutInflater.from(parent.context)
             .inflate(R.layout.row_date, parent, false)
+
         return ViewHolder(view)
     }
 
+    /**
+     * This is an Important Method for RecyclerView, called onBindViewHolder.
+     *
+     * Called by RecyclerView to display data at position,
+     * second parameter is the position
+     */
     override fun onBindViewHolder(holder: ViewHolder, position: Int) {
-        holder.bind(data[position])
+        holder.bindDate(data[position])
```

Comment:
It's probably better to have one `bind` function on ViewHolder. It's easy to miss calling multiple functions like this when refactoring in the future. Additionally, if the RecyclerView ever needed to support multiple view holder types, this code would quickly become unwieldy.